lib/BinaryNode.cpp

```diff
@@ -39,22 +39,20 @@ BinaryNode::BinaryNode(void* p, qore_size_t size) : SimpleValueQoreNode(NT_BINAR
 }
 
 BinaryNode::~BinaryNode() {
-    if (ptr)
+    if (ptr) {
         free(ptr);
+    }
 }
 
 void BinaryNode::clear() {
-    if (len) {
-        assert(ptr);
+    // NOTE: must check 'ptr', len may be 0 with memory allocated
+    if (ptr) {
         free(ptr);
```

Comment:
this can lead to an undefined behavior, consider
```
void *x; // undefined, may point to something
BinaryNode node(x, 0);
// dtor of BinaryNode or ::clear() will call free to undefined pointer
```